tests/test_api_users.py

```diff
@@ -323,4 +352,196 @@ def test_03_create_update_delete_unicode_user(self):
+                                           data={"realm": "realm1"},
+                                           headers={'Authorization': self.at}):
+            res = self.app.full_dispatch_request()
+            self.assertTrue(res.status_code == 200, res)
+            result = res.json.get("result")
+            self.assertTrue(result.get("status"), res.data)
+            additional_attribute_found = False
+            # check in the user list for the username=cornelius
+            for user in result.get("value"):
+                if user.get("username") == "cornelius":
+                    self.assertEqual(user.get("newattribute"), "newvalue")
+                    additional_attribute_found = True
+            self.assertTrue(additional_attribute_found)
+
+        # Now we search for the one explicit user
+        with self.app.test_request_context('/user/',
+                                           method='GET',
+                                           data={"realm": "realm1",
+                                                 "username": "cornelius"},
+                                           headers={'Authorization': self.at}):
+            res = self.app.full_dispatch_request()
+            self.assertTrue(res.status_code == 200, res)
+            result = res.json.get("result")
+            self.assertTrue(result.get("status"), res.data)
+            # check in the user list for the username=cornelius
+            self.assertEqual(len(result.get("value")), 1)
+            self.assertEqual(result.get("value")[0].get("newattribute"), "newvalue")
+
+        # The additional attribute should also be returned, if the user authenticates successfully.
+        init_token({"serial": "SPASS1", "type": "spass", "pin": "test"}, user=User("cornelius", self.realm1))
+        set_policy(name="POL1", scope=SCOPE.AUTHZ, action=ACTION.ADDUSERINRESPONSE)
+        with self.app.test_request_context('/validate/check',
+                                           method='POST',
+                                           data={"user": "cornelius@realm1",
+                                                 "pass": "test"},
+                                           headers={'Authorization': self.at}):
+            res = self.app.full_dispatch_request()
+            self.assertTrue(res.status_code == 200, res)
+            result = res.json.get("result")
+            details = res.json.get("detail")
+            user_data = details.get("user")
+            self.assertIn("newattribute", user_data)
+            self.assertEqual(user_data.get("newattribute"), "newvalue")
+        remove_token("SPASS1")
+        delete_policy("POL1")
+
+        # Now we delete the additional user attribute
+        set_policy("custom_attr", scope=SCOPE.ADMIN,
+                   action="{0!s}=*".format(ACTION.DELETE_USER_ATTRIBUTES))
+        with self.app.test_request_context('/user/attribute/newattribute/cornelius/realm1',
```

Comment:
Please also test deleting an attribute without the policy and deleting an unknown attribute.

Reply:
Will check and ...

will do.

Reply:
Done in be5753ee